### PR TITLE
Register missing WebRTCDataChannelJS type

### PR DESCRIPTION
**Pull request by akien-mga**
Rebased #55044 on the `master` branch as that PR was wrongly targeting `3.4`.

### CODE REVIEW

File: modules/webrtc/webrtc_peer_connection_js.cpp [hunk 1]
```diff
@@ -57,7 +57,7 @@ void WebRTCPeerConnectionJS::_on_error(void *p_obj) {
 
 void WebRTCPeerConnectionJS::_on_data_channel(void *p_obj, int p_id) {
 	WebRTCPeerConnectionJS *peer = static_cast<WebRTCPeerConnectionJS *>(p_obj);
-	peer->emit_signal(SNAME("data_channel_received"), Ref<WebRTCDataChannelJS>(new WebRTCDataChannelJS(p_id)));
+	peer->emit_signal(SNAME("data_channel_received"), Ref<WebRTCDataChannel>(new WebRTCDataChannelJS(p_id)));
```
Comment (Faless) [inline modules/webrtc/webrtc_peer_connection_js.cpp:60]: I think replacing `new` with `memnew` was also okay (it should help track memory leak IIRC).
The only real problem with the original PR was registering it to ClassDB.

Comment (akien-mga) [reply modules/webrtc/webrtc_peer_connection_js.cpp:60]: Oh right I missed that when resolving conflict.

Comment (akien-mga) [reply modules/webrtc/webrtc_peer_connection_js.cpp]: Should be fixed.

**Review by Faless — APPROVED** [review]

**Comment by akien-mga** [comment]
Cherry-picked for 3.5.

**Pull request merged**